src/generate/apiModel.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+
+import {
+  resolveApiModel,
+  parseRamlFile,
+  getAllDataTypes,
+} from "../common/parser";
+
+import { Name } from "../common/structures/name";
+import { ApiMetadata } from "./apiMetadata";
+import { RestApi } from "../download/exchangeTypes";
+import path from "path";
+import fs from "fs-extra";
+import { ramlToolLogger } from "../common/logger";
+import _ from "lodash";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export class ApiModel extends ApiMetadata {
+  model: model.document.Document;
+  dataTypes: model.domain.CustomDomainProperty[] = [];
+
+  metadata: RestApi;
+  main: string;
+  path: string;
+  /**
+   *Creates an instance of ApiModel.
+   * @param {string} name - Name of the model
+   * @param {string} filepath - Root path of the api with an exchange.json OR raml file to read in.
+   * @param {ApiMetadata[]} [children=([] = [])] - NYI - Any children that the API might have,
```

Comment:
Is there a possibility that we use this in future. just thinking it may be confusing when reading/using the code

Reply:
I think there is.  Mostly wanted to have this to keep the constructor consistent with the parent.